models/migrations/v1_23/v305.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package v1_23 //nolint
+
+import (
+	"code.gitea.io/gitea/modules/timeutil"
+
+	"xorm.io/xorm"
+)
+
+func AddRepositoryLicenses(x *xorm.Engine) error {
+	type RepoLicense struct {
+		ID          int64 `xorm:"pk autoincr"`
+		RepoID      int64 `xorm:"UNIQUE(s) INDEX NOT NULL"`
+		CommitID    string
+		License     string             `xorm:"VARCHAR(50) UNIQUE(s) INDEX NOT NULL"`
```

Comment:
1. Is `50` chars enough? Why not 200 or 255 (as default)
2. Does the `INDEX` really work or it is just redundant and costs unnecessary CPU&storage?

Reply:
1. 8f5a70dc7020c4d4f5578d7e41b4b1132d32dc84
2. It is just redundant. I will remove it in this PR, we can add it later if necessary.
![image](https://github.com/user-attachments/assets/7326e6a9-78f5-41da-8bf9-1764573c55fc)